lib/spring/application.rb

```diff
@@ -1,9 +1,17 @@
 require "spring/boot"
 require "set"
 require "pty"
+require "spring/platform"
 
 module Spring
   class Application
+    if Spring.fork?
+      require 'spring/application/fork_strategy'
+      include ForkStrategy
+    else
+      require 'spring/application/pool_strategy'
+      include PoolStrategy
+    end
```

Comment:
Why use modules here? Why not have `ForkedApplication` and `PooledApplication` subclasses of the same abstract class?